pkg/apis/application/v1alpha1/types.go

```diff
@@ -327,7 +330,32 @@ func (in *ApplicationSourceHelm) AddFileParameter(p HelmFileParameter) {
 
 // IsZero Returns true if the Helm options in an application source are considered zero
 func (h *ApplicationSourceHelm) IsZero() bool {
-	return h == nil || (h.Version == "") && (h.ReleaseName == "") && len(h.ValueFiles) == 0 && len(h.Parameters) == 0 && len(h.FileParameters) == 0 && h.Values == "" && !h.PassCredentials && !h.IgnoreMissingValueFiles && !h.SkipCrds
+	return h == nil || (h.Version == "") && (h.ReleaseName == "") && len(h.ValueFiles) == 0 && len(h.Parameters) == 0 && len(h.FileParameters) == 0 && h.ValuesIsEmpty() && !h.PassCredentials && !h.IgnoreMissingValueFiles && !h.SkipCrds
+}
+
+func (h *ApplicationSourceHelm) SetValuesString(value string) error {
+	h.Values = &runtime.RawExtension{Raw: []byte(value)}
+	return nil
+}
+
+func (h *ApplicationSourceHelm) ValuesYAML() []byte {
+	if h.Values == nil {
+		return []byte{}
+	}
+	b, err := yaml.JSONToYAML(h.Values.Raw)
+	if err != nil {
+		// This should be impossible, because rawValue isn't set directly.
```

Comment:
Isn't rawValue populated with arbitrary values coming from the `Application` resource? Wouldn't it be possible to have that value as an invalid json that would cause an error in this call? If so, it is better not swallowing the error and returning it instead.

Reply:
the way this is structured, we couldn't surface an error here anyway
in my tests, I wasn't able to put arbitrary data, as kubernetes validations would prohibit it